channeldb/db.go

```diff
@@ -616,6 +617,55 @@ func (d *DB) FetchClosedChannel(chanID *wire.OutPoint) (*ChannelCloseSummary, er
 	return chanSummary, nil
 }
 
+// FetchClosedChannelForID queries for a channel close summary using the
+// channel ID of the channel in question.
+func (d *DB) FetchClosedChannelForID(cid lnwire.ChannelID) (
```

Comment:
This as is scales with the number of closed channels, which will increase for all time. Instead, we can use a different algorithm, which will be constant-time ish, as in we know the hard upper bound

  1. Fetch the entire set of channel point candidates for a channel ID via `GenPossibleOutPoints`
  2. Run through each of these, and attempt to read the key from the `closedChannelBucket`

This should terminate in 1, and sometimes two iterations. Only once people start doing this like splicing or funding coin joins will this start to swell. Even then, it should be well below 1000 or so iterations.

Reply:
Actually scratch that, I properly see the seek!